wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/Utility/MessageSendFailureView.swift

```diff
@@ -0,0 +1,102 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireCommonComponents
+import Down
+
+final class MessageSendFailureView: UIView {
+
+    // MARK: Properties
+    var tapHandler: ((UIButton) -> Void)?
+
+    private let stackView = UIStackView(axis: .vertical)
+    private let titleLabel = WebLinkTextView()
+    private let retryButton: IconButton = {
+        let button = InviteButton()
+        button.titleLabel?.font = FontSpec.buttonSmallSemibold.font!
+        button.contentEdgeInsets = UIEdgeInsets(top: 8, left: 8, bottom: 8, right: 8)
+        button.setTitle(L10n.Localizable.Content.System.FailedtosendMessage.retry, for: .normal)
```

Comment:
@KaterinaWire How about modifying our current init in InviteButton so we can pass all those properties during init? I think it makes sense. What do you think?

Reply:
Yes, it makes sense. Please take a look